src/sempy_labs/perf_lab/_simulated_etl.py

```
@@ -0,0 +1,494 @@
+        StructField("DatasourceName", StringType(), nullable=False),
+        StructField("DatasourceWorkspace", StringType(), nullable=False),
+        StructField("DatasourceType", StringType(), nullable=False),
+        StructField("SourceTableName", StringType(), nullable=False),
+        StructField("SourceFormat", StringType(), nullable=False),
+        StructField("SourceType", StringType(), nullable=False),
+        StructField("SourceLocation", StringType(), nullable=False),
+    ])
+    rows = []
+
+    for row in test_definitions.dropDuplicates(['TargetWorkspace', 'TargetDataset','DatasourceName','DatasourceWorkspace','DatasourceType']).collect():
+        target_dataset = row['TargetDataset']
+        target_workspace = row['TargetWorkspace']
+        data_source_name = row['DatasourceName']
+        data_source_workspace = row['DatasourceWorkspace']
+        data_source_type = row['DatasourceType']
+
+        # Skip this row if the data_source_type is invalid.
+        if not data_source_type == "Lakehouse":
+            print(f"{icons.red_dot} Invalid data source type '{data_source_type}' detected. Ignoring this row. Please review your test definitions.")
+            continue 
+
+        # Skip this row if the target semantic model is not defined.
+        if target_dataset == "" or target_dataset is None:
+            print(f"{icons.red_dot} No test semantic model specifed as the target dataset. Ignoring this row. Please review your test definitions.")
+            continue
+
+        # Skip this row if the data_source_name is not defined.
+        if data_source_name == "" or data_source_name is None:
+            print(f"{icons.red_dot} No data source found for test semantic model '{target_dataset}'. Ignoring this row. Please review your test definitions.")
+            continue
+
+        try:
+            (target_workspace_name, target_workspace_id) = resolve_workspace_name_and_id(workspace=target_workspace)
+        except:
```

catch specific exception, otherwise we mask errors

---

Thanks. Replaced with (WorkspaceNotFoundException, ValueError).